client/api_client.go

```diff
@@ -568,15 +584,16 @@ func (c *client) generateURL(bucketName string, objectName string, relativePath
 func (c *client) signRequest(req *http.Request) error {
 	// use offChainAuth if OffChainAuthOption is set
 	if c.offChainAuthOption != nil {
-		authStr := c.OffChainAuthSign()
-		// set auth header
-		req.Header.Set(types.HTTPHeaderAuthorization, authStr)
 		req.Header.Set("X-Gnfd-User-Address", c.defaultAccount.GetAddress().String())
 		req.Header.Set("X-Gnfd-App-Domain", c.offChainAuthOption.Domain)
+		unsignedMsg := httplib.GetMsgToSignInGNFD1Auth(req)
+		authStr := c.OffChainAuthSign(unsignedMsg)
+		// set auth header
+		req.Header.Set(types.HTTPHeaderAuthorization, authStr)
```

Comment:
may need to add httplib.httplib.Gnfd1Eddsa to the authStr

Reply:
It is already added, in  c.OffChainAuthSign method.